clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -609,8 +609,8 @@ def ConditionOp : CIR_Op<"condition", [
 //===----------------------------------------------------------------------===//
 
 def YieldOp : CIR_Op<"yield", [ReturnLike, Terminator,
-                               ParentOneOf<["IfOp", "ScopeOp", "WhileOp",
-                                            "ForOp", "DoWhileOp"]>]> {
+                               ParentOneOf<["IfOp", "TernaryOp", "ScopeOp",
```

Comment:
What is the ordering of this list?  Does it make sense to group them in some way?  Or perhaps alpha order?

Reply:
I put it next to `IfOp` since both are about branching. But maybe lexicographic ordering is more sensible. The incubator does neither FWIW.

Reply:
Probably not actionable on this patch, but something for @bcardosolopes to consider?  Perhaps an incubator patch to give SOME meaningful sort here :)